app/forms/instance-create.tsx

```diff
@@ -177,10 +186,33 @@ export function CreateInstanceForm() {
   const form = useForm({ defaultValues })
   const { control, setValue } = form
 
-  const imageInput = useWatch({ control: control, name: 'bootDiskSource' })
-  const image = allImages.find((i) => i.id === imageInput)
+  const siloImageSource = useWatch({ control: control, name: 'siloImageSource' })
+  const projectImageSource = useWatch({ control: control, name: 'projectImageSource' })
+  const diskSource = useWatch({ control: control, name: 'diskSource' })
+  const bootDiskSourceType = useWatch({ control: control, name: 'bootDiskSourceType' })
+  const bootDiskSource = useWatch({ control: control, name: 'bootDiskSource' })
+  const bootDiskSize = useWatch({ control: control, name: 'bootDiskSize' })
+  const image = allImages.find((i) => i.id === bootDiskSource)
   const imageSize = image?.size ? Math.ceil(image.size / GiB) : undefined
 
+  useEffect(() => {
+    bootDiskSourceType === 'siloImage' && setValue('bootDiskSource', siloImageSource)
+    bootDiskSourceType === 'projectImage' && setValue('bootDiskSource', projectImageSource)
+    bootDiskSourceType === 'disk' && setValue('bootDiskSource', diskSource)
```

Comment:
these are redundant with the ones in `onValueChange`, but like I said there, I don't think either is necessary